tests/server/query_params/test_sort.py

```diff
@@ -98,3 +100,73 @@ def test_datetime_desc(get_good_response, structures):
     ]
     assert last_modified_list == expected_last_modified
+
+
+def test_unknown_field_errors(check_error_response, structures):
+    """ If any completely unknown field is provided, check 400: Bad Request is returned. """
+    limit = 5
+    request = f"/structures?sort=field_that_does_not_exist&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field 'field_that_does_not_exist'",
+    )
+
+    request = f"/structures?sort=field_that_does_not_exist,-other_field_that_does_not_exist&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown fields 'field_that_does_not_exist', 'other_field_that_does_not_exist'",
+    )
+
+    request = f"/structures?sort=field_that_does_not_exist,nelements&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field 'field_that_does_not_exist'",
+    )
+
+    # case 6: non-existent provider field
+    request = f"/structures?sort=_exmpl_provider_field_that_does_not_exist,nelements&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field '_exmpl_provider_field_that_does_not_exist'",
+    )
+
+
+def test_unknown_field_prefixed(get_good_response, structures):
+    """ If any other-provider-specific fields are requested, return a warning but still sort. """
+    limit = 5
+    request = f"/structures?sort=_exmpl3_field_that_does_not_exist,nelements&page_limit={limit}"
+    data = structures.collection.find(sort=[("nelements", 1)], limit=limit)
+    expected_nelements = [_["nelements"] for _ in data]
+
+    with pytest.warns(FieldNotRecognized):
+        response = get_good_response(request)
+
+    expected_detail = (
+        "Unable to sort on unknown field '_exmpl3_field_that_does_not_exist'"
+    )
+    assert len(response["meta"]["warnings"]) == 1
+    assert response["meta"]["warnings"][0]["detail"] == expected_detail
+    assert response["meta"]["warnings"][0]["title"] == "FieldNotRecognized"
+
+    nelements_list = [
+        struct.get("attributes", {}).get("nelements") for struct in response["data"]
+    ]
+    assert nelements_list == expected_nelements
+
+    # case 5: only prefixed fields
+    request = f"/structures?sort=-_exmpl2_field_that_does_not_exist,_exmpl3_other_field&page_limit={limit}"
+    with pytest.warns(FieldNotRecognized):
+        response = get_good_response(request)
+
+    expected_detail = "Unable to sort on unknown fields '_exmpl2_field_that_does_not_exist', '_exmpl3_other_field'"
```

Comment:
The same may be possible here, i.e., using `match=expected_detail` after reversing the order here.